pkg/network/network.go

```diff
@@ -94,10 +103,16 @@ func NewConfigFromConfigMap(configMap *corev1.ConfigMap) (*Config, error) {
 	} else {
 		nc.IstioOutboundIPRanges = normalizedIpr
 	}
+
 	if ingressClass, ok := configMap.Data[DefaultClusterIngressClassKey]; !ok {
 		nc.DefaultClusterIngressClass = IstioIngressClassName
 	} else {
 		nc.DefaultClusterIngressClass = ingressClass
 	}
+
+	// If not in the config data then it'll be empty, which is ok, we'll
+	// just use the default value at that point
+	nc.RouteTemplate = configMap.Data[RouteTemplateKey]
```

Comment:
Why not implement the defaulting behavior right here?

Reply:
You mean set it to name.ns.domain instead of "" ?
Because if someone sets the value in the configMap to "", or just erases it, then we'll still need to recover from that later on and I'd prefer to have the defaulting logic in only one spot.